### ISSUE TITLE
BadJpqlGrammarException: no viable alternative at input

### ISSUE THREAD

**Issue by m-ignatov** [spring-projects/spring-data-jpa#3093]
Spring Data JPA throws BadJpqlGrammarException: Line 1:139 no viable alternative at input 'L.forceUnlockTime<:time'; Bad JPQL grammar [UPDATE Lock L SET L.isLocked = TRUE, L.forceUnlockTime = :forceUnlockTime WHERE L.isLocked = FALSE OR L.forceUnlockTime < :time]
```
@Query("UPDATE Lock L SET L.isLocked = TRUE, L.forceUnlockTime = :forceUnlockTime WHERE L.isLocked = FALSE OR L.forceUnlockTime < :time")
int acquireLock(@Param("forceUnlockTime") Date forceUnlockTime, @Param("time") Date time);
```  
We are using the latest Spring Boot BOM 3.1.2 with Spring Data JPA 3.1.2.

**Comment by gregturn** [comment]
Merged to `main`, backported to `3.1.x`.

**Comment by quaff** [comment]
I'm wondering if https://github.com/spring-projects/spring-data-jpa/pull/3010 could resolve such problem permanently.
